callouts/java/service-callout/src/main/java/example/BasicCalloutServer.java

```diff
@@ -0,0 +1,167 @@
+                ImmutableMap.of("response-header", "added-response", "c", "d").entrySet(),  // Headers to add
+                ImmutableList.of("foo"),  // Headers to remove
+                false,  // Do not clear route cache
+                null  // No append action
+        );
+    }
+
+    /**
+     * Handles the request body by appending a custom suffix to the existing body.
+     * <p>
+     * The method appends "-added-body" to the original HTTP request body without clearing it.
+     *
+     * @param processingResponseBuilder the builder used to construct the {@link ProcessingResponse}.
+     * @param body the {@link HttpBody} object representing the original request body.
+     */
+    @Override
+    public void onRequestBody(ProcessingResponse.Builder processingResponseBuilder, HttpBody body) {
+        // Modify the body by appending "-added-body" and ensure no clearBody action
+        addBodyMutations(processingResponseBuilder.getRequestBodyBuilder(), body.getBody().toStringUtf8() + "-added-body", null, null);
+    }
+
+    /**
+     * Handles the response body by replacing it with a static value.
+     * <p>
+     * The method completely replaces the original response body with "body replaced".
+     *
+     * @param processingResponseBuilder the builder used to construct the {@link ProcessingResponse}.
+     * @param body the {@link HttpBody} object representing the original response body.
+     */
+    @Override
+    public void onResponseBody(ProcessingResponse.Builder processingResponseBuilder, HttpBody body) {
+        // Replace the body with "body replaced"
+        addBodyMutations(processingResponseBuilder.getResponseBodyBuilder(), "body replaced", null, null);
+    }
+
+    /**
+     * Main method to start the gRPC callout server with a custom configuration
+     * using the {@link ServiceCallout.Builder}.
+     * <p>
+     * This method initializes the server with default or custom configurations,
+     * starts the server, and keeps it running until manually terminated.
+     * The server processes incoming gRPC requests for HTTP manipulations.
+     * </p>
+     *
+     * <p>Usage:</p>
+     * <pre>{@code
+     * ServiceCallout.Builder builder = new ServiceCallout.Builder()
+     *     .setIp("111.222.333.444")       // Customize IP
+     *     .setPort(8443)                  // Set the port for secure communication
+     *     .setEnableInsecurePort(true)    // Enable an insecure communication port
+     *     .setServerThreadCount(4);       // Set the number of server threads
+     * }</pre>
+     *
+     * @param args Command-line arguments, not used in this implementation.
+     * @throws Exception If an error occurs during server startup or shutdown.
+     */
+    public static void main(String[] args) throws Exception {
+        // Create a builder for ServiceCallout with custom configuration
+        BasicCalloutServer server = new BasicCalloutServer.Builder()
+                .setHealthCheckPort(8000)
```

Comment:
Since we want the default configuration to work with the default callout setup workflow this should be removed by default.